gyp/pylib/gyp/input.py

```diff
@@ -236,7 +236,10 @@ def LoadOneBuildFile(build_file_path, data, aux_data, includes, is_target, check
             # But since node-gyp produces ebcdic files, do not use that mode.
             build_file_contents = open(build_file_path, "r").read()
         else:
-            build_file_contents = open(build_file_path, "rU").read()
+            if (sys.version_info[0] >= 3) and (sys.version_info[1] >= 7):
+                build_file_contents = open(build_file_path, 'rU', encoding="utf8").read()
+            else:
+                build_file_contents = open(build_file_path, 'rU').read()
```

Comment:
I seriously doubt this works... https://softwareengineering.stackexchange.com/questions/298677/why-is-universal-newlines-mode-deprecated-in-python

Reply:
I dont know. It was before me.